packages/std/src/memory.rs

```diff
@@ -11,7 +11,10 @@ use crate::errors::{Error, NullPointer};
 #[repr(C)]
 pub struct Region {
     pub offset: u32,
-    pub len: u32,
+    /// The number of bytes available in this region
+    pub capacity: u32,
+    /// The number of bytes used in this region
+    pub length: u32,
 }
 
 /// alloc is the same as external allocate, but designed to be called internally
```

Comment:
Ideally `alloc` will return a Region with `capacity = size`, and `length = 0`. This is what we send - empty data with capacity. Maybe it can no longer reuse `release_buffer`, which assumes there is data there?

Reply:
you can use https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity

`Vec::with_capacity(size)` returns with length 0

Reply:
Implemented in a33061baf3cb28cf525ac37f815e96f65a2adf23 and 61d2581dc9db350d30080d30462e828eb1ebacf0. The later causes a massive drop in gas usage (see 547be8c3a86d81e398f9c5c86ac1ff36adc1cf97). Very nice find.

Reply:
Main difference is that allocated memory is not zeroed.